python/python/glide/routes.py

```diff
@@ -48,6 +48,24 @@ def __init__(self, slot_type: SlotType, slot_id: int) -> None:
         self.slot_id = slot_id
 
 
+class ByAddressRoute(Route):
+    def __init__(self, host: str, port: Optional[int] = None) -> None:
+        """Routes a request to a node by its DNS address
+
+        Args:
+            host (str): The DNS name of the node. If `port` is not provided, should be in the `{DNS}:{port}` format.
+            port (Optional[int]): The port the node communicates on.
+        """
+        super().__init__()
+        if port == None:
+            split = host.split(":")
+            host = split[0]
+            port = int(split[1])
```

Comment:
It's a potential error, we should catch this exception and raise appropriate client error

Reply:
May be use a 3rd lib which does all checks for us?
Consider that a user may give two different ports or port out of range.